### PR TITLE
onedpl add fetch.lua

### PR DESCRIPTION
**Pull request by Altina-oz**
tested with TBB backend

and it should work for omp backend

I wonder if TBB links/linkdirs should be explicitly added or these can be implicitly inferred after adding package deps ("openmp")?

### CODE REVIEW

**Comment by Altina-oz** [comment]
io.replace(...pstl/...):

c.f. 

https://github.com/llvm/llvm-project/commit/3b9a1bb1af90db9472340ef2122d3855eb9ba3fc#diff-4c6821476cefc699b801f5fdbeda3341e3c64626dcf39a79621ea02031bdd50eL113

https://github.com/oneapi-src/oneDPL/issues/1602

**Comment by Altina-oz** [comment]
> io.replace(...pstl/...):
> 
> c.f.
> 
> [llvm/llvm-project@3b9a1bb#diff-4c6821476cefc699b801f5fdbeda3341e3c64626dcf39a79621ea02031bdd50eL113](https://github.com/llvm/llvm-project/commit/3b9a1bb1af90db9472340ef2122d3855eb9ba3fc#diff-4c6821476cefc699b801f5fdbeda3341e3c64626dcf39a79621ea02031bdd50eL113)
> 
> [oneapi-src/oneDPL#1602](https://github.com/oneapi-src/oneDPL/issues/1602)

solution c.f.

https://github.com/oneapi-src/oneDPL/pull/1551

https://github.com/oneapi-src/oneDPL/pull/1551/files

File: packages/o/onedpl/xmake.lua [hunk 1]
```diff
@@ -5,17 +5,36 @@ package("onedpl")
     set_description("oneAPI DPC++ Library")
 
     add_urls("https://github.com/oneapi-src/oneDPL/archive/refs/tags/oneDPL-$(version)-release.tar.gz")
+    add_urls("https://github.com/oneapi-src/oneDPL/archive/refs/tags/oneDPL-$(version).tar.gz")
```
Comment (waruqi) [inline packages/o/onedpl/xmake.lua:8]: 应该放开头，否则每次都有一次 404 failed

File: packages/o/onedpl/fetch.lua [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import("lib.detect.find_path")
+import("lib.detect.find_library")
+
+function _find_package(package, opt)
+    local paths = {
+	      "$(env DPL_ROOT)",
```
Comment (waruqi) [inline packages/o/onedpl/fetch.lua:6]: 缩进不对

File: packages/o/onedpl/fetch.lua [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        "$(env ONEAPI_ROOT)/dpl/latest"
+    }
+    -- find library
+    local result = {links = {}, linkdirs = {}, includedirs = {}}
+    local backend = package:config("backend")
+    
+    if backend == "tbb" or backend == "dpcpp" then
+        package:add("deps", "tbb")
```
Comment (waruqi) [inline packages/o/onedpl/fetch.lua:14]: 不是这么玩的， fetch 里面不应该 add deps

File: packages/o/onedpl/fetch.lua [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    elseif backend == "omp" then
+        package:add("deps", "openmp")
+    end
+    if package:is_plat("windows") then
+        package:add("cxxflags", "/Zc:__cplusplus")
```
Comment (waruqi) [inline packages/o/onedpl/fetch.lua:19]: fetch 里面，也不能 add 这种，任何对包的修改都不行

Comment (Altina-oz) [reply packages/o/onedpl/fetch.lua]: 噢也就是只能xmake.lua

File: packages/o/onedpl/xmake.lua (hunk 1 above)
Comment (Altina-oz) [reply packages/o/onedpl/xmake.lua:7]: 如果以后出了新的，它是-release.tar.gz而不是.rc?.tar.gz,是不是还要把顺序换回来o.O我在想或者直接把release写到版本号得了

Comment (waruqi) [reply packages/o/onedpl/xmake.lua:7]: 参考 sqlite3 的包，写 version function

File: packages/o/onedpl/fetch.lua [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    local result = {links = {}, linkdirs = {}, includedirs = {}}
+
+    -- find include
+    local includepath = find_path(path.join("oneapi","dpl","algorithm"), paths, {suffixes = "include"})
+    if includepath then
+        table.insert(result.includedirs, includepath)
+    end
+	
+    if #result.includedirs > 0  then
+        local version_file = path.join(includepath, "oneapi","dpl","pstl","onedpl_config.h")
```
Comment (waruqi) [inline packages/o/onedpl/fetch.lua:19]: 还有这，逗号两边空格

**Pull request merged**